src/winforms/toga_winforms/widgets/textinput.py

```diff
@@ -8,18 +8,25 @@ class TextInput(Widget):
     def create(self):
         self.native = WinForms.TextBox()
         self.native.Multiline = False
+        self.native.TextChanged += self.winforms_onTextChanged
+        self.native.Leave += self.winforms_onFocusLost
 
     def set_readonly(self, value):
         self.native.ReadOnly = value
 
     def set_placeholder(self, value):
-        self.native.Text = self.interface.placeholder
+        if self.interface.placeholder and not self.interface.value:
+            self.native.Click += self.winforms_Click
+            self.native.Text = self.interface.placeholder
+            self.native.ForeColor = Color.FromName('GRAY')
```

Comment:
The colors should probably come from a system definition, rather than being hardcoded GRAY/BLACK - that way, if the user has customized a "dark" theme, (so they have white text on black background) they'll still be able to see their text.